benchmarks/Kestrel.Performance/InMemoryTransportBenchmark.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    public class InMemoryTransportBenchmark
+    {
+        private const int _plaintextExpectedResponseLength = 132;
+        private const int _plaintextPipelinedExpectedResponseLength = _plaintextExpectedResponseLength * RequestParsingData.Pipelining;
+
+        private IWebHost _host;
+        private InMemoryConnection _connection;
+
+        [GlobalSetup(Target = nameof(Plaintext) + "," + nameof(PlaintextPipelined))]
```

Comment:
Why do you have Target here but not on cleanup?

Reply:
And why have it here at all?

Reply:
Future-proofing for scenarios other than Plaintext.  Setup is specific to Plaintext, but Cleanup just disposes the host so it would apply to all scenarios.